### PR TITLE
PartyType enum

### PR DESCRIPTION
**Pull request by Hunter2807**
## Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by Hunter2807** [comment]
Well, I don't really know much about the docs part, so you will have to add the `PartyType` class to the docs, so that people can know the supported party types

File: disnake/abc.py [hunk 1]
```diff
@@ -1071,7 +1071,7 @@ async def create_invite(
 
             .. versionadded:: 2.0
 
-        target_application_id:: Optional[:class:`int`]
+        target_application_id:: Optional[:class:`.PartyType`]
```
Comment (m1raynee) [inline disnake/abc.py:1074]:
```suggestion
        target_application: Optional[:class:`.PartyType`]
```

**Review by m1raynee — CHANGES_REQUESTED** [review]
Also take a loop at abc.py line 1102. This must be 
```py
target_application_id=target_application.value if target_application else None
```

File: disnake/abc.py [hunk 2]
```diff
@@ -1099,7 +1099,7 @@ async def create_invite(
             unique=unique,
             target_type=target_type.value if target_type else None,
             target_user_id=target_user.id if target_user else None,
-            target_application_id=target_application_id,
+            target_application_id=target_application.value if target_application else None,
```
Comment (thesadru) [inline disnake/abc.py:1102]: Actually, `try_enum_to_int` would be much better here.

**Review by thesadru — CHANGES_REQUESTED** [review]

File: disnake/abc.py (hunk 2 above)
Comment (Hunter2807) [reply disnake/abc.py:1102]: Okay, lemme fix it

Comment (thesadru) [reply disnake/abc.py:1102]: while you're at that do it for `target_type` too

**Pull request merged**